src/main/scala/com/microsoft/partnercatalyst/fortis/spark/sinks/cassandra/CassandraEventSchema.scala

```diff
@@ -0,0 +1,128 @@
+package com.microsoft.partnercatalyst.fortis.spark.sinks.cassandra
+
+import java.util.Date
+
+import com.microsoft.partnercatalyst.fortis.spark.dto.FortisEvent
+import com.microsoft.partnercatalyst.fortis.spark.sinks.cassandra.dto._
+import java.text.Collator
+import java.util.Locale
+
+import com.microsoft.partnercatalyst.fortis.spark.analyzer.timeseries.{Period, PeriodType}
+import com.microsoft.partnercatalyst.fortis.spark.transforms.gender.GenderDetector.{Female, Male}
+
+object CassandraEventSchema {
+  def apply(item: FortisEvent, batchid: String): Event = {
+    Event(
+      pipelinekey = item.details.pipelinekey,
+      externalsourceid = item.details.externalsourceid,
+      computedfeatures = Utils.getFeature(item),
+      eventtime = item.details.eventtime,
+      batchid = batchid,
+      eventlangcode = item.analysis.language.orNull,
+      eventid = item.details.eventid,
+      insertiontime = new Date().getTime,
+      body = item.details.body,
+      sourceurl = item.details.sourceurl,
+      title = item.details.title)
+  }
+}
+
+object CassandraPopularPlaces {
+  def apply(item: EventBatchEntry): Seq[PopularPlaceAggregate] = {
+    for {
+      kw <- Utils.getConjunctiveTopics(Option(item.computedfeatures.keywords))
+      location <- item.computedfeatures.places
+      periodType <- Utils.getCassandraPeriodTypes
+    } yield PopularPlaceAggregate(
+      pipelinekey = item.pipelinekey,
+      centroidlat = location.centroidlat,
+      centroidlon = location.centroidlon,
+      placeid = location.placeid,
+      periodstartdate = Period(item.eventtime.getTime, periodType).startTime(),
+      periodenddate = Period(item.eventtime.getTime, periodType).endTime(),
+      periodtype = periodType.periodTypeName,
+      period = periodType.format(item.eventtime.getTime),
+      externalsourceid = item.externalsourceid,
+      mentioncount = item.computedfeatures.mentions,
```

Comment:
Summarizing conversation over the phone. We can simplify this by using the `counter` column type.

This has a number of advantages, most prominently:
1) We don't have to read the current value for every write.
2) We avoid race conditions as Cassandra counters are guaranteed to be eventually consistent.
3) The code will be simpler as aggregations are now just single writes.

In essence I propose a schema like this:

```cql
CREATE TABLE myAggregation_data (
  key_col1 text,
  key_col2 text,
  data_col1 text,
  counter_pointer text,
  PRIMARY KEY (key_col1, key_col2)
);

CREATE TABLE myAggregation_counters (
  id text,
  value1 counter,
  value2 counter,
  length counter,
  PRIMARY KEY (id)
);
```

We can insert into this schema like so:

```cql
INSERT INTO myAggregation_data VALUES (
  -- same as we insert now
) IF NOT EXISTS;

UPDATE myAggregation_counters
SET value1 = value1 + 23
  , value2 = value2 + 12
  , length = length + 1
WHERE id = ?;
```

We can then read back the average values like so:

```js
const dataQuery = `
// same query as now, only excluding the mentioncount/sentiment_avg columns
`.trim();

const dataParams = [
  // from the graphql args, as we query now
];

const counterQuery = `
SELECT * FROM myAggregation_counters
WHERE id IN ?
`.trim();

cassandraConnector.execute(dataQuery, dataParams)
.then(rows => {
  const counterIds = rows.map(row => row.counter_pointer);
  cassandraConnector.execute(counterQuery, [counterIds])
  .then(counters => {
    const avg_value1 = mean(counters.map(counter =>
      counter.value1 / counter.length));

    // do something with avg_value1 here
  });
});
```

Limitations of this approach:
- For each read, we make two trips to Cassandra -- this is OK because we have many more writes than reads.
- Limited precision as we can no longer store floats, counters are just integers. This is okay because we can just truncate-multiply the sentiment float value like 0.754 => 754.